xla/service/gpu/runtime/nccl_collective_permute_thunk.cc

```diff
@@ -133,6 +161,12 @@ NcclCollectivePermuteStartThunk::NcclCollectivePermuteStartThunk(
 absl::Status NcclCollectivePermuteStartThunk::Initialize(
     const InitializeParams& params) {
   TF_RETURN_IF_ERROR(NcclCollectiveThunk::Initialize(params));
+  const se::Platform* platform = params.executor->GetPlatform();
+  CHECK(platform != nullptr);
+  device_count_ = platform->VisibleDeviceCount();
```

Comment:
There could be several "visible" devices that aren't actually being targeted by the running XLA instance.  Is it always ok to assume that NCCL collectives are using all visible devices?

Reply:
good point, i see that in pjrt client it always uses visible devices to build the distributed topology. But if xla decides to not use all visible devices, this logic in the pr wont be correct. Do you know if there's any way to get the set of targeted devices?

Reply:
Maybe use LocalClient::device_count or Backend::device_count?  I think either of those would be correct.  (Backend::device_count is I think derived from LocalClient::device_count)

Reply:
OK seems like those options are not easily accessible in gpu executable. I added a device count property in ExecutableRunOptions, and the thunk will just read the value from there.